satpy/readers/native_msg.py

```diff
@@ -270,75 +280,94 @@ def get_area_extent(self, dsid):
         data15hd = self.header['15_DATA_HEADER']
         sec15hd = self.header['15_SECONDARY_PRODUCT_HEADER']
 
-        if dsid.name != 'HRV':
+        # check for Earth model as this affects the north-south and
+        # west-east offsets
+        # section 3.1.4.2 of MSG Level 1.5 Image Data Format Description
+        earth_model = data15hd['GeometricProcessing']['EarthModel'][
+            'TypeOfEarthModel']
+        if earth_model == 2:
+            ns_offset = 0
+            we_offset = 0
+        elif earth_model == 1:
+            ns_offset = -0.5
+            we_offset = 0.5
+            if dsid.name == 'HRV':
+                ns_offset = -1.5
+                we_offset = 1.5
+        else:
+            raise NotImplementedError(
+                'Unrecognised Earth model: {}'.format(earth_model)
+            )
 
-            # following calculations assume grid origin is south-east corner
-            # section 7.2.4 of MSG Level 1.5 Image Data Format Description
-            origins = {0: 'NW', 1: 'SW', 2: 'SE', 3: 'NE'}
-            grid_origin = data15hd['ImageDescription'][
-                "ReferenceGridVIS_IR"]["GridOrigin"]
-            if grid_origin != 2:
-                raise NotImplementedError(
-                    'Grid origin not supported number: {}, {} corner'
-                    .format(grid_origin, origins[grid_origin])
-                )
+        # Calculations assume grid origin is south-east corner
+        # section 7.2.4 of MSG Level 1.5 Image Data Format Description
+        if dsid.name == 'HRV':
+            reference_grid = 'ReferenceGridHRV'
+        else:
+            reference_grid = 'ReferenceGridVIS_IR'
+
+        origins = {0: 'NW', 1: 'SW', 2: 'SE', 3: 'NE'}
+        grid_origin = data15hd['ImageDescription'][
+            reference_grid]['GridOrigin']
+        if grid_origin != 2:
+            msg = 'Grid origin not supported number: {}, {} corner'.format(
+                grid_origin, origins[grid_origin]
+            )
+            raise NotImplementedError(msg)
+
+        # When dealing with HRV channel and full disk, area extent is
+        # in two pieces
+        if (dsid.name == 'HRV') and self.mda['is_full_disk']:
+            # TODO: Implement HRV full disk area_extent
+            # NotImplementedError does not catch this, it must
+            # be used elsewhere already
+            msg = 'HRV full disk area extent not implemented.'
+            raise RuntimeError(msg)
+
+        # Otherwise area extent is in one piece, corner points are
+        # the same as for VISIR channels, HRV channel is having
+        # three times the amount of columns and rows
+        else:
 
-            center_point = 3712/2
+            if dsid.name == 'HRV':
+                center_point = 11136/2
```

Comment:
Maybe use global constants like `HRV_NUM_COLUMNS`, `VISIR_NUM_COLUMNS`, and same for rows, instead of plain numbers for clarity?